### PR TITLE
DBZ-4415 oplog mode is not supported for MongoDB 5+

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-4415

### CODE REVIEW

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnector.java [hunk 1]
```diff
@@ -223,6 +225,32 @@ public Config validate(Map<String, String> connectorConfigs) {
...
+                    if (versionDocument != null) {
+                        final String versionString = versionDocument.getString("version");
+                        if (versionString != null) {
+                            final String[] versionComponents = versionString.split("\\.");
+                            if (versionComponents.length > 0) {
+                                try {
+                                    final int majorVersion = Integer.parseInt(versionComponents[0]);
+                                    final ConfigValue captureModeValue = results
+                                            .get(MongoDbConnectorConfig.CAPTURE_MODE.name());
+                                    final String captureMode = config.getString(MongoDbConnectorConfig.CAPTURE_MODE);
```
Comment (gunnarmorling) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnector.java:241]: Wouldn't it be better to construct a `MongoDbConnectorConfig()` and call the typed accessor for this prop?

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnector.java [hunk 2]
```diff
@@ -223,6 +225,32 @@ public Config validate(Map<String, String> connectorConfigs) {
...
+                                    if (majorVersion >= 5
+                                            && CaptureMode.OPLOG.getValue().equalsIgnoreCase(captureMode)) {
+                                        captureModeValue.addErrorMessage(
+                                                "'oplog' capture mode is not supported for MongoDB 5 and newer");
```
Comment (gunnarmorling) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnector.java:245]:
```suggestion
                                                "The 'oplog' capture mode is not supported for MongoDB 5 and newer; Please use 'change_streams'  or 'change_streams_update_full' instead");
```

**Review by gunnarmorling — APPROVED** [review]
LGTM, some minor comments inline.

**Comment by jpechane** [comment]
@gunnarmorling Comments addressed.

**Pull request merged**